src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -388,6 +391,23 @@ namespace
 
             shading_system.register_closure(name(), id(), params, 0, 0);
         }
+
+        static void convert_closure(
+            CompositeEmissionClosure&   composite_closure,
+            const void*                 osl_params,
+            const Color3f&              weight,
+            const float                 max_weight_component)
+        {
+            InputValues* values =
+                composite_closure.add_closure<InputValues>(
+                    id(),
+                    weight,
+                    max_weight_component);
+
+            const float rcp_max_weight_component = 1.0f / max_weight_component;
+            values->m_radiance = weight * rcp_max_weight_component;
```

Comment:
Looks like you can just divide by `max_weight_component` and get rid of `rcp_max_weight_component`.